swarm.go

```diff
@@ -539,30 +538,30 @@ func (s *Swarm) APIs() []rpc.API {
 		apis = append(apis, s.ps.APIs()...)
 	}
 
+	if s.config.SwapEnabled {
+		// Swap public API
+		apis = append(apis, s.swap.APIs()...)
+	}
+
 	return apis
 }
 
-// SetChequebook ensures that the local checquebook is set up on chain.
-func (s *Swarm) SetChequebook(ctx context.Context) error {
-	err := s.config.Swap.SetChequebook(ctx, s.backend, s.config.Path)
-	if err != nil {
-		return err
-	}
-	log.Info(fmt.Sprintf("new chequebook set (%v): saving config file, resetting all connections in the hive", s.config.Swap.Contract.Hex()))
-	return nil
+// DeploySwap ensures that Swap is set up on chain.
+func (s *Swarm) DeploySwap(ctx context.Context) error {
+	return s.swap.Deploy(ctx, s.backend, s.config.Path)
 }
 
 // RegisterPssProtocol adds a devp2p protocol to the swarm node's Pss instance
 func (s *Swarm) RegisterPssProtocol(topic *pss.Topic, spec *protocols.Spec, targetprotocol *p2p.Protocol, options *pss.ProtocolParams) (*pss.Protocol, error) {
 	return pss.RegisterProtocol(s.ps, topic, spec, targetprotocol, options)
 }
 
-// serialisable info about swarm
+// Info represents serialisable info about swarm
```

Comment:
you mean `represents the current Swarm node's configuration`

Reply:
fixed